doc/api/fs.md

```diff
@@ -1549,6 +1549,23 @@ Synchronous version of [`fs.readFile`][]. Returns the contents of the `file`.
 If the `encoding` option is specified then this function returns a
 string. Otherwise it returns a buffer.
 
+_Note: The behavior of `fs.readFileSync()` is platform specific. As such,
+reading a directory on macOS and Linux and Windows - see example
+below - will return an error. In contrast, on FreeBSD, contents of
+the file will be returned._
```

Comment:
If that is what happens, then yes, exactly.
BTW: that's strange, how do you know when one file ends and another begins... 🤔